requirements/_tests.txt

```diff
@@ -15,4 +15,5 @@ requests <=2.31.0
 fire <=0.5.0
 
 cloudpickle >1.3, <=3.0.0
-scikit-learn >=1.1.1, <1.4.0
+scikit-learn >=1.1.1, <1.3.0; python_version < "3.9"
+scikit-learn >=1.4.0, <1.5.0; python_version >= "3.9"
```

Comment:
if 1.4.0 breaks it, then the change would be
```suggestion
scikit-learn >=1.1.1, <1.4.0; python_version < "3.9"
scikit-learn >=1.4.0, <1.5.0; python_version >= "3.9"
```

Reply:
@mathematicalmichael  Thank you for your comment.

I intentionally skipped version 1.3 because the sklearn 1.3 has a bug breaking `zero_division` test.
As of Python 3.8, this is a summary:
- sklearn 1.4 does not support py3.8 --> NG
- sklearn 1.3 support py3.8 but has a bug breaking tests --> NG
- sklearn 1.2 support py3.8 and does not have the bug --> OK
I think sklearn 1.2 is the latest version that can pass the tests in this PR.

Reply:
@mathematicalmichael Sorry for the confusion. I want to follow up on my explanation.

There are two problems now:

1. sklearn 1.3.x has a bug related to zero_division that will break this PR test.
2. sklearn >= 1.4.x that fixed the bug does not support python <= 3.8

Your suggestion `scikit-learn >=1.1.1, <1.4.0; python_version < "3.9"` would break CI test for python3.8  because the condition allows installing sklearn==1.3.x that has the zero_division bug.

Fortunately, the latest sklearn==1.4 fixed the bug and passed all other torchmetrics test cases. 

If I understand correctly, the original condition `scikit-learn >=1.1.1, <1.4.0` means CI should pass one of the versions from 1.1.1 to 1.3.x and does not mean that CI passes all of them. 

So, I think we can update the condition to `scikit-learn >=1.4.0, <1.5.0` for the `python_version >= 3.9` if there is no dependency problem.

As for `python <= 3.8`, I had to set `scikit-learn < 1.3`, but it is a subset of the original condition and should not conflict with it.